nexus/src/external_api/http_entrypoints.rs

```diff
@@ -8884,6 +8890,66 @@ async fn role_view(
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
 
+// Current user
+
+/// Fetch the user associated with the current session
+#[endpoint {
+   method = GET,
+   path = "/v1/current-user",
+   tags = ["hidden"],
+}]
+pub async fn current_user_view_v1(
+    rqctx: RequestContext<Arc<ServerContext>>,
+) -> Result<HttpResponseOk<views::User>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let handler = async {
+        // We don't care about authentication method, as long as they are authed
+        // as _somebody_. We could restrict this to session auth only, but it's
+        // not clear what the advantage would be.
```

Comment:
I know you're just copying this comment, but I think we can lose this. Maybe just something along the lines of "we report information about the user irrespective the authentication method"

Reply:
Doesn’t need a comment at all, really. This was early when we didn’t really know how we’d think about authz and session. At this point it goes without saying that whatever operations are authorized will go through, otherwise they’ll fail, and the calling code isn’t responsible for deciding any of that.